Add "coq.env.query-extra-dep" external predicate

phikal opened the pull request:
As discussed on [Zulip](https://coq.zulipchat.com/#narrow/stream/253928-Elpi-users-.26-devs/topic/Resolving.20Extra.20Dependencies/near/396087392), this would allow for a predicate to resolve the real path behind a `Extra Dependency`.  As seen in the patch, this just amounts to wrapping `ComExtraDeps.query_extra_dep` in the right way.

gares commented on src/coq_elpi_builtins.ml:
```diff
@@ -2325,6 +2325,15 @@ denote the same x as before.|};
             Some (c, np + na))))))),
   DocAbove);
 
+  MLCode(Pred("coq.env.query-extra-dep",
```
`coq.env` is for accessing the logical environment.
Please move this code just before 
```
-- Datatypes conversions --------------------------------------------
```
maybe in a dedicated section, say `-- Extra Dependency ------------------------`.

Also I'd call it `coq.extra-dep`

gares reviewed (changes requested):
Thanks for this PR.
There are a few things to improve before merging, see the comments.

You should also add a line to the ChangeLog.md file

phikal commented:
Enrico Tassi ***@***.***> writes:

> @gares requested changes on this pull request.
>
> Thanks for this PR.
> There are a few things to improve before merging, see the comments.
>
> You should also add a line to the ChangeLog.md file

Done.

>> @@ -0,0 +1,7 @@
> +From elpi Require Import elpi.
> +
> +From unreleased Extra Dependency "elpi_elaborator.elpi" as elab.
> +
> +Elpi Query lp:{{ coq.env.query-extra-dep "elab" File }}.
> +
> +Fail Elpi Query lp:{{ coq.env.query-extra-dep "foo" File }}.
>
> Please add the test file to _CoqProject.test.
>
> This file is a good test, but the API returns None instead of raising
> `No_clause`, so this will fail once it runs.

I've replaced it with an example that matches (some _).

>> @@ -2325,6 +2325,15 @@ denote the same x as before.|};
>              Some (c, np + na))))))),
>    DocAbove);
>  
> +  MLCode(Pred("coq.env.query-extra-dep",
>
> `coq.env` is for accessing the logical environment.
> Please move this code just before 
> ```
> -- Datatypes conversions --------------------------------------------
> ```
> maybe in a dedicated section, say `-- Extra Dependency ------------------------`.
>
> Also I'd call it `coq.extra-dep`

Fine with me, I've also changed this.

gares commented on src/coq_elpi_builtins.ml:
```diff
@@ -3884,6 +3884,15 @@ Supported attributes:
      state, !: s, [])),
   DocAbove);
 
+  MLCode(Pred("coq.extra-dep",
```
Please add an LPDoc line with something like
```
-- Extra Dependencies -------------
```

gares commented on coq-builtin.elpi:
```diff
@@ -901,6 +901,10 @@ external pred coq.env.projections i:inductive, o:list (option constant).
 external pred coq.env.primitive-projections i:inductive, 
                                             o:list (option (pair projection int)).
 
+% [coq.env.query-extra-dep Identifier File Name] Resolve the file name of an
```
Please regenerate this file

gares reviewed (changes requested):
A few more nits

phikal commented:
Sorry, missed those.

phikal commented:
Do you have a schedule when you are planning to publish a release with these changes?

gares commented:
Before the end of November
